packages/api/src/sequelize/migrations/2025-01-19_00_create-hie_directory_view-with-fts.ts

```diff
@@ -0,0 +1,67 @@
+import { QueryTypes } from "sequelize";
+import type { Migration } from "..";
+
+const tableName = "cq_directory_entry_new";
+const hieViewName = "hie_directory_view";
+const cqViewName = "cq_directory_entry_view";
+const columnName = "search_criteria";
+
+const createSearchCriteriaColumnSql = `
+alter table ${tableName}
+add ${columnName} tsvector
+generated always as	(
+	to_tsvector('english', coalesce(id, '')) || ' ' ||
+	to_tsvector('english', coalesce(name, '')) || ' ' ||
+	to_tsvector('english', coalesce(root_organization, '')) || ' ' ||
+	to_tsvector('english', coalesce(address_line, '')) || ' ' ||
+	to_tsvector('english', coalesce(city, '')) || ' ' ||
+	to_tsvector('english', coalesce(state, '')) || ' ' ||
+	to_tsvector('english', coalesce(zip, ''))
+) stored;
+`;
+
+const createGinIndexSql = `
+CREATE INDEX IF NOT EXISTS ${tableName}_${columnName}_idx 
+ON ${tableName} 
+USING gin(${columnName});
+`;
+
+const dropCqViewSql = `DROP VIEW IF EXISTS ${cqViewName};`;
+const createCqViewSql = `CREATE VIEW ${cqViewName} AS SELECT * from ${tableName};`;
+const createHieViewSql = `CREATE VIEW ${hieViewName} AS SELECT * from cq_directory_entry_view;`;
+
+// Use 'Promise.all' when changes are independent of each other
+export const up: Migration = async ({ context: queryInterface }) => {
+  await queryInterface.sequelize.transaction(async transaction => {
+    await Promise.all(
+      [
+        createSearchCriteriaColumnSql,
+        createGinIndexSql,
+        dropCqViewSql,
+        createCqViewSql,
+        createHieViewSql,
+      ].map(sql =>
```

Comment:
Those changes are dependent on each other, right? Thus, we should not use `Promise.all` (see comment above the `up` fn).

Reply:
Oh duh yeah. For some reason I thought it was gonna run them in serial when i wrote this, though that's obviously the antithesis of why one uses Promise.all, great callout